app/uk/gov/hmrc/personaldetailsvalidation/audit/EventsSender.scala

```diff
@@ -61,7 +61,12 @@ private[personaldetailsvalidation] class EventsSender @Inject()(platformAnalytic
   }
 
   private implicit class PersonalDetailsOps(target: PersonalDetails) {
-    def hasSameNinoSuffixAs(other: PersonalDetails): Boolean = target.nino.value.last == other.nino.value.last
+    def hasSameNinoSuffixAs(other: PersonalDetails): Boolean = {
+      (target, other) match {
+        case (first : PersonalDetailsNino, second : PersonalDetailsNino) => first.nino.value == second.nino.value
+        case _ => false
```

Comment:
This case can end up sending GA event `success_nino_suffix_different_from_cid ` when postcode is provided as user input and it is a successful matching.
We should not be sending GA event in this cases.

Reply:
Isn't this part of TSV-952? I think we would need input from the BAs to determine the exact GA events that need to be created.